factory_client.py

```diff
@@ -143,7 +170,12 @@ def get_target_system_image(self, target: Target, out_dir: str, progress: Progre
         else:
             logger.info('Target system image has been already downloaded: {}'.format(extracted_image_file_path))
 
-        return extracted_image_file_path
+        release_resp = requests.get(os_release_url, headers=self._auth_headers)
+        if release_resp.ok:
+            release_info = dict([line.split('=') for line in release_resp.content.decode().splitlines()])
+        else:
```

Comment:
i can see that for now this will be the normal case, but in the future it seems like we'd be silently ignoring an important error?

Reply:
> but in the future it seems like we'd be silently ignoring an important error?

I am not sure if it's critical error and a CI build should be stopped in this case.
For the "assemble" run triggered within the LmP job it can be kind of critical error provided that this PR https://github.com/foundriesio/meta-lmp/pull/496 is included into a given LmP version. But, it will take some time for all customers to move to the LmP version that includes `os-release` dumping.
Even if all customers move to the os-release dumping version of LmP they might use `fioctl targets image` to preload Apps into some older version of LmP.
Maybe we can just add an info log informing about the presence or absence of `os-release`?

Reply:
I added an info log if `os-release`  is missing and an error message if `os-release` is present but a parse error happen. Just not sure if we need to (re-)throw exception and exit the CI job/run in this case or just log an error message. On the one hand it's not really critical because the assemble run can do its job without a release info and simply choose a type of Apps to preload based on an explicitly defined type in a user factory (factory-config.yml) and preload `compose` apps if it's not explicitly defined.
Another approach is to exit/fail the CI job if a parse error happens and a type of Apps is not explicitly defined.